### PR TITLE
CircleAvatar docs

### LINKED ISSUE #6731: Avatar Circle
When using this widget it always say that need a child. If the child is null the app crash.
I don't know if the chid field it has to be not null or what is happening.

### PR DESCRIPTION
**Pull request by Hixie**
Fixes https://github.com/flutter/flutter/issues/6731

### CODE REVIEW

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/material/circle_avatar.dart [hunk 1]
```diff
@@ -14,22 +15,38 @@ import 'theme.dart';
...
     this.backgroundImage,
     this.radius: 20.0
   }) : super(key: key);
 
   /// The widget below this widget in the tree.
+  ///
+  /// Typically a [Text] or [Image] widget. For performance reasons, images are
+  /// not clipped by default. If your image is not pre-clipped to a circle
+  /// shape, consider preprocessing the image before the application loads it.
+  /// If that is not possible, consider using a [ClipOval] widget.
```
Comment (abarth) [inline packages/flutter/lib/src/material/circle_avatar.dart]: Maybe recommend using a `backgroundImage`

**Comment by Hixie** [comment]
@abarth I reworked this per our discussion. PTAL.

File: packages/flutter/lib/src/material/circle_avatar.dart [hunk 2]
```diff
@@ -4,32 +4,58 @@
...
+///  * [Chip], for representing users or concepts in long form.
+///  * [ListItem], which combines an icon (such as a [CircleAvatar]) with some
+///    text in a horizontal row.
 ///  * <https://material.google.com/components/chips.html#chips-contact-chips>
 class CircleAvatar extends StatelessWidget {
   /// Creates a circle that represents a user.
   CircleAvatar({
     Key key,
-    this.child,
+    @required this.child,
```
Comment (abarth) [inline packages/flutter/lib/src/material/circle_avatar.dart]: This shouldn't be required anymore.

**Comment by abarth** [comment]
LGTM

**Comment by Hixie** [comment]
Removed @required, will land if Travis agrees.

**Pull request merged**